tests/noise_model_test.py

```diff
@@ -0,0 +1,722 @@
+        qubit_list[0]: 0, qubit_list[1]: 0, qubit_list[2]: 1}
+    assert pauli.phase == 0
+
+    pauli.CX(qubit_list[1], qubit_list[2])
+    assert pauli.X_list == {
+        qubit_list[0]: 0, qubit_list[1]: 0, qubit_list[2]: 0}
+    assert pauli.Z_list == {
+        qubit_list[0]: 0, qubit_list[1]: 1, qubit_list[2]: 1}
+    assert pauli.phase == 0
+
+    pauli.H(qubit_list[1])
+    assert pauli.X_list == {
+        qubit_list[0]: 0, qubit_list[1]: 1, qubit_list[2]: 0}
+    assert pauli.Z_list == {
+        qubit_list[0]: 0, qubit_list[1]: 0, qubit_list[2]: 1}
+    assert pauli.phase == 0
+
+    pauli.S(qubit_list[1])
+    assert pauli.X_list == {
+        qubit_list[0]: 0, qubit_list[1]: 1, qubit_list[2]: 0}
+    assert pauli.Z_list == {
+        qubit_list[0]: 0, qubit_list[1]: 1, qubit_list[2]: 1}
+    assert pauli.phase == 1
+
+    pauli.CX(qubit_list[1], qubit_list[2])
+    assert pauli.X_list == {
+        qubit_list[0]: 0, qubit_list[1]: 1, qubit_list[2]: 1}
+    assert pauli.Z_list == {
+        qubit_list[0]: 0, qubit_list[1]: 0, qubit_list[2]: 1}
+    assert pauli.phase == 1
+
+    pauli.S(qubit_list[2])
+    assert pauli.X_list == {
+        qubit_list[0]: 0, qubit_list[1]: 1, qubit_list[2]: 1}
+    assert pauli.Z_list == {
+        qubit_list[0]: 0, qubit_list[1]: 0, qubit_list[2]: 0}
+    assert pauli.phase == 2
+
+    pauli.CX(qubit_list[1], qubit_list[0])
+    assert pauli.X_list == {
+        qubit_list[0]: 1, qubit_list[1]: 1, qubit_list[2]: 1}
+    assert pauli.Z_list == {
+        qubit_list[0]: 0, qubit_list[1]: 0, qubit_list[2]: 0}
+    assert pauli.phase == 2
+
+    pauli.S(qubit_list[0])
+    assert pauli.X_list == {
+        qubit_list[0]: 1, qubit_list[1]: 1, qubit_list[2]: 1}
+    assert pauli.Z_list == {
+        qubit_list[0]: 1, qubit_list[1]: 0, qubit_list[2]: 0}
+    assert pauli.phase == 3
+
+    pauli.H(qubit_list[0])
+    assert pauli.X_list == {
+        qubit_list[0]: 1, qubit_list[1]: 1, qubit_list[2]: 1}
+    assert pauli.Z_list == {
+        qubit_list[0]: 1, qubit_list[1]: 0, qubit_list[2]: 0}
+    assert pauli.phase == 1
+
+
+def test_to_from_qps():
+
+    qubits = [Qubit(name=f'reg_name_{i}', index=0) for i in range(5)]
+    paulis = [Pauli.I, Pauli.Y, Pauli.X, Pauli.Y, Pauli.Z]
+    qubit_pauli_string = QubitPauliString(
+        qubits=qubits, paulis=paulis,
+    )
+    stab = QermitPauli.from_qubit_pauli_string(qubit_pauli_string)
+    stab_qps, stab_phase = stab.qubit_pauli_string
+    assert stab_qps == qubit_pauli_string
+    assert stab_phase == 1 + 0j
+
+
+def test_is_measureable():
+
+    stab = QermitPauli(
+        Z_list=[1, 0, 0],
+        X_list=[1, 0, 1],
+        qubit_list=[Qubit(name='A', index=0), Qubit(name='B', index=0), Qubit(name='A', index=1)]
+    )
+    assert stab.is_measureable(qubit_list=[Qubit(name='A', index=1)])
+    assert stab.is_measureable(qubit_list=[Qubit(name='A', index=0)])
+    assert not stab.is_measureable(qubit_list=[Qubit(name='B', index=0)])
```

Comment:
I like to add an if name == main at the end of the file and call all the files, & add `->None` to each function definition.

Reply:
I have done this for the test that do not require a temporary directory to be set up. I'm sure that it's possible to run those tests too but I've not been able to figure out how. Do you have any experience with this?